internal/local/structure_v2.go

```diff
@@ -601,19 +607,35 @@ func writeDataSources(rootDir string, dataSources []DataSourceStructure) error {
 			); err != nil {
 				return err
 			}
-			if err := WriteFile(
-				filepath.Join(ruleDir, FileSchema.String()),
-				0666,
-				bytes.NewReader(dataSchema),
-			); err != nil {
-				return err
+
+			schema := rule[NameSchema]
+			if schema != nil {
```

Comment:
let's use the `if schema, ok := rule[NameSchema]; ok {` syntax here (we avoided it previously b/c we always wanted to replace `nil` with `map[string]interface{}{}` which i suppose is no longer the case)